src/main/java/com/tempoiq/UpsertResponse.java

```diff
@@ -0,0 +1,102 @@
+package com.tempoiq;
+
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpEntity;
+import org.apache.http.util.EntityUtils;
+import com.tempoiq.json.Json;
+
+public class UpsertResponse {
+  private HashMap<String, DeviceStatus> statuses;
+
+  public UpsertResponse(HashMap<String, DeviceStatus> response) {
+    this.statuses = response;
+  }
+
+  public boolean wasSuccessful() {
+    for (Map.Entry<String, DeviceStatus> entry : statuses.entrySet()) {
+      if (entry.getValue().getSuccess() == false) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  public boolean wasPartiallySuccessful() {
+    int failures = 0;
+    Set<Map.Entry<String, DeviceStatus>> entries = this.statuses.entrySet();
+    for (Map.Entry<String, DeviceStatus> entry : entries) { 
+      if (entry.getValue().getSuccess() == false) {
+        failures += 1;
+      }
+    }
+    if (failures < entries.size()) {
+      return true;
+    }
+    return false;
+  }
+
+  private HashMap<String, DeviceStatus> filterByState(DeviceStatus.DeviceState state) {
+    HashMap<String, DeviceStatus> results = new HashMap<String, DeviceStatus>();
+    for (Map.Entry<String, DeviceStatus> entry : this.statuses.entrySet()) { 
+      if (entry.getValue().getState() == state) {
+        results.put(entry.getKey(), entry.getValue());
+      }
+    }
+    return results;
+  }
+
+  public HashMap<String, DeviceStatus> getExisting() {
+    return filterByState(DeviceStatus.DeviceState.EXISTING);
+  }
+
+  public HashMap<String, DeviceStatus> getModified() {
+    return filterByState(DeviceStatus.DeviceState.MODIFIED);
+  }
+
+  public HashMap<String, DeviceStatus> getCreated() {
+    return filterByState(DeviceStatus.DeviceState.CREATED);
+  }
+
+  public HashMap<String, DeviceStatus> getFailures() {
+    HashMap<String, DeviceStatus> results = new HashMap<String, DeviceStatus>();
+    for (Map.Entry<String, DeviceStatus> entry : this.statuses.entrySet()) { 
+      if (entry.getValue().getSuccess() == false) {
+        results.put(entry.getKey(), entry.getValue());
+      }
+    }
+    return results;
+  }
+  
+  public static UpsertResponse make(HttpResponse response) throws java.io.IOException {
```

Comment:
Objects not toys.  Stupid swypos
On Mar 13, 2015 3:58 PM, "Aaron Brenzel" <aaron.brenzel@tempoiq.com> wrote:

> It's used by Result to construct toys dynamically through reflection so I
> had to match the method name and signature it expects
> On Mar 13, 2015 3:57 PM, "James Clemer" <notifications@github.com> wrote:
>
>> In src/main/java/com/tempoiq/UpsertResponse.java
>> <https://github.com/TempoIQ/tempoiq-java/pull/19#discussion_r26422299>:
>>
>> > +
>> > +  public HashMap<String, DeviceStatus> getCreated() {
>> > +    return filterByState(DeviceStatus.DeviceState.CREATED);
>> > +  }
>> > +
>> > +  public HashMap<String, DeviceStatus> getFailures() {
>> > +    HashMap<String, DeviceStatus> results = new HashMap<String, DeviceStatus>();
>> > +    for (Map.Entry<String, DeviceStatus> entry : this.statuses.entrySet()) {
>> > +      if (entry.getValue().getSuccess() == false) {
>> > +        results.put(entry.getKey(), entry.getValue());
>> > +      }
>> > +    }
>> > +    return results;
>> > +  }
>> > +
>> > +  public static UpsertResponse make(HttpResponse response) throws java.io.IOException {
>>
>> Why isn't this just a constructor overload?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/TempoIQ/tempoiq-java/pull/19/files#r26422299>.
>>
>